app/App/Panel/Main/Account/Requests/TransactionRequest/TxApproval/approvals/TokenSpend/index.js

```diff
@@ -35,9 +35,11 @@ function nFormat (n, digits = 2)  {
 class TokenSpend extends React.Component {
   constructor (...args) {
     super(...args)
+    const { approval: { data }, revoke } = this.props
 
-    this.decimals = this.props.approval.data.decimals || 0
-    this.requestedAmount = '0x' + new BigNumber(this.props.approval.data.amount).integerValue().toString(16)
+    this.decimals = data.decimals || 0
+    this.requestedAmount = '0x' + new BigNumber(data.amount).integerValue().toString(16)
+    this.revoke = !!revoke
```

Comment:
this doesnt need to be an instance variable at all does it? we can just reference the props directly everywhere we need it

Reply:
It does if we want to make it a boolean - accessing directly can be undefined unless we explicitly set false in the parent component

Reply:
Here's the linting rule I am referring to: 
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md

Reply:
I think just letting it be falsy is totally fine but if you want to get the same effect you're going for you can just use `const revoke = !!this.props.revoke` at the top of the render method. the props are already set on the instance by React and we dont need another instance variable w. the same data.

Reply:
I'll update for falsy approach, not actually bothered about a consistent exact type unless we need to address it because of conversion to TS which we're not doing in this PR